```
./dom/Range.cpp:183:14: error: no member named 'is_eq' in namespace 'std'
    if (std::is_eq(ordering))
        ~~~~~^
./dom/Range.cpp:251:14: error: no member named 'is_eq' in namespace 'std'
```

--- a/Source/WebCore/dom/BoundaryPoint.cpp
+++ b/Source/WebCore/dom/BoundaryPoint.cpp
@@ -29,9 +29,9 @@
 
 namespace WebCore {
 
-template PartialOrdering treeOrder<Tree>(const BoundaryPoint&, const BoundaryPoint&);
-template PartialOrdering treeOrder< ShadowIncludingTree >(const BoundaryPoint&, const BoundaryPoint&);
-template PartialOrdering treeOrder<ComposedTree>(const BoundaryPoint&, const BoundaryPoint&);
+template std::partial_ordering treeOrder<Tree>(const BoundaryPoint&, const BoundaryPoint&);
+template std::partial_ordering treeOrder<ShadowIncludingTree>(const BoundaryPoint&, const BoundaryPoint&);
+template std::partial_ordering treeOrder<ComposedTree>(const BoundaryPoint&, const BoundaryPoint&);
 
 std::optional<BoundaryPoint> makeBoundaryPointBeforeNode(Node& node)
 {
@@ -64,40 +64,29 @@ static bool isOffsetBeforeChild(ContainerNode& container, unsigned offset, Node&
     return false;
 }
 
-// FIXME: Once we move to C++20, replace with the C++20 <=> operator.
-// FIXME: This could return std::strong_ordering if we had that, or the equivalent.
-static PartialOrdering order(unsigned a, unsigned b)
-{
-    if (a < b)
-        return PartialOrdering::less;
-    if (a > b)
-        return PartialOrdering::greater;
-    return PartialOrdering::equivalent;
-}
-
-template<TreeType treeType> PartialOrdering treeOrder(const BoundaryPoint& a, const BoundaryPoint& b)
+template<TreeType treeType> std::partial_ordering treeOrder(const BoundaryPoint& a, const BoundaryPoint& b)
 {
     if (a.container.ptr() == b.container.ptr())
-        return order(a.offset, b.offset);
+        return a.offset <=> b.offset;
 
     for (auto ancestor = b.container.ptr(); ancestor; ) {
         auto nextAncestor = parent<treeType>(*ancestor);
         if (nextAncestor == a.container.ptr())
-            return isOffsetBeforeChild(*nextAncestor, a.offset, *ancestor) ? PartialOrdering::less : PartialOrdering::greater;
+            return isOffsetBeforeChild(*nextAncestor, a.offset, *ancestor) ? std::strong_ordering::less : std::strong_ordering::greater;
         ancestor = nextAncestor;
     }
 
     for (auto ancestor = a.container.ptr(); ancestor; ) {
         auto nextAncestor = parent<treeType>(*ancestor);
         if (nextAncestor == b.container.ptr())
-            return isOffsetBeforeChild(*nextAncestor, b.offset, *ancestor) ? PartialOrdering::greater : PartialOrdering::less;
+            return isOffsetBeforeChild(*nextAncestor, b.offset, *ancestor) ? std::strong_ordering::greater : std::strong_ordering::less;
         ancestor = nextAncestor;
     }
 
     return treeOrder<treeType>(a.container, b.container);
 }
 
-PartialOrdering treeOrderForTesting(TreeType type, const BoundaryPoint& a, const BoundaryPoint& b)
+std::partial_ordering treeOrderForTesting(TreeType type, const BoundaryPoint& a, const BoundaryPoint& b)
 {
     switch (type) {
     case Tree:
@@ -108,7 +97,7 @@ PartialOrdering treeOrderForTesting(TreeType type, const BoundaryPoint& a, const
         return treeOrder<ComposedTree>(a, b);
     }
     ASSERT_NOT_REACHED();
-    return PartialOrdering::unordered;
+    return std::partial_ordering::unordered;
 }
 
 TextStream& operator<<(TextStream& stream, const BoundaryPoint& boundaryPoint)

--- a/Source/WebCore/dom/BoundaryPoint.h
+++ b/Source/WebCore/dom/BoundaryPoint.h
@@ -42,14 +42,14 @@ bool operator==(const BoundaryPoint&, const BoundaryPoint&);
 
 WTF::TextStream& operator<<(WTF::TextStream&, const BoundaryPoint&);
 
-template<TreeType = Tree> PartialOrdering treeOrder(const BoundaryPoint&, const BoundaryPoint&);
+template<TreeType = Tree> std::partial_ordering treeOrder(const BoundaryPoint&, const BoundaryPoint&);
 
 WEBCORE_EXPORT std::optional<BoundaryPoint> makeBoundaryPointBeforeNode(Node&);
 WEBCORE_EXPORT std::optional<BoundaryPoint> makeBoundaryPointAfterNode(Node&);
 BoundaryPoint makeBoundaryPointBeforeNodeContents(Node&);
 BoundaryPoint makeBoundaryPointAfterNodeContents(Node&);
 
-WEBCORE_EXPORT PartialOrdering treeOrderForTesting(TreeType, const BoundaryPoint&, const BoundaryPoint&);
+WEBCORE_EXPORT std::partial_ordering treeOrderForTesting(TreeType, const BoundaryPoint&, const BoundaryPoint&);
 
 inline BoundaryPoint::BoundaryPoint(Ref<Node>&& container, unsigned offset)
     : container(WTFMove(container))

--- a/Source/WebCore/dom/Node.cpp
+++ b/Source/WebCore/dom/Node.cpp
@@ -2785,35 +2785,35 @@ static bool isSiblingSubsequent(const Node& siblingA, const Node& siblingB)
     return false;
 }
 
-template<TreeType treeType> PartialOrdering treeOrder(const Node& a, const Node& b)
+template<TreeType treeType> std::partial_ordering treeOrder(const Node& a, const Node& b)
 {
     if (&a == &b)
-        return PartialOrdering::equivalent;
+        return std::partial_ordering::equivalent;
     auto result = commonInclusiveAncestorAndChildren<treeType>(a, b);
     if (!result.commonAncestor)
-        return PartialOrdering::unordered;
+        return std::partial_ordering::unordered;
     if (!result.distinctAncestorA)
-        return PartialOrdering::less;
+        return std::partial_ordering::less;
     if (!result.distinctAncestorB)
-        return PartialOrdering::greater;
+        return std::partial_ordering::greater;
     bool isShadowRootA = result.distinctAncestorA->isShadowRoot();
     bool isShadowRootB = result.distinctAncestorB->isShadowRoot();
     if (isShadowRootA || isShadowRootB) {
         if (!isShadowRootB)
-            return PartialOrdering::less;
+            return std::partial_ordering::less;
         if (!isShadowRootA)
-            return PartialOrdering::greater;
+            return std::partial_ordering::greater;
         ASSERT_NOT_REACHED();
-        return PartialOrdering::unordered;
+        return std::partial_ordering::unordered;
     }
-    return isSiblingSubsequent(*result.distinctAncestorA, *result.distinctAncestorB) ? PartialOrdering::less : PartialOrdering::greater;
+    return isSiblingSubsequent(*result.distinctAncestorA, *result.distinctAncestorB) ? std::partial_ordering::less : std::partial_ordering::greater;
 }
 
-template PartialOrdering treeOrder<Tree>(const Node&, const Node&);
-template PartialOrdering treeOrder<ShadowIncludingTree>(const Node&, const Node&);
-template PartialOrdering treeOrder<ComposedTree>(const Node&, const Node&);
+template std::partial_ordering treeOrder<Tree>(const Node&, const Node&);
+template std::partial_ordering treeOrder<ShadowIncludingTree>(const Node&, const Node&);
+template std::partial_ordering treeOrder<ComposedTree>(const Node&, const Node&);
 
-PartialOrdering treeOrderForTesting(TreeType type, const Node& a, const Node& b)
+std::partial_ordering treeOrderForTesting(TreeType type, const Node& a, const Node& b)
 {
     switch (type) {
     case Tree:
@@ -2824,7 +2824,7 @@ PartialOrdering treeOrderForTesting(TreeType type, const Node& a, const Node& b)
         return treeOrder<ComposedTree>(a, b);
     }
     ASSERT_NOT_REACHED();
-    return PartialOrdering::unordered;
+    return std::partial_ordering::unordered;
 }
 
 TextStream& operator<<(TextStream& ts, const Node& node)

--- a/Source/WebCore/dom/Node.h
+++ b/Source/WebCore/dom/Node.h
@@ -31,6 +31,7 @@
 #include "StyleValidity.h"
 #include "TaskSource.h"
 #include "TreeScope.h"
+#include <compare>
 #include <wtf/CompactPointerTuple.h>
 #include <wtf/CompactUniquePtrTuple.h>
 #include <wtf/FixedVector.h>
@@ -745,38 +746,17 @@ private:
 
 bool connectedInSameTreeScope(const Node*, const Node*);
 
-// Designed to be used the same way as C++20 std::partial_ordering class.
-// FIXME: Consider putting this in a separate header.
-// FIXME: Once we can require C++20, replace with std::partial_ordering.
-class PartialOrdering {
-public:
-    static const PartialOrdering less;
-    static const PartialOrdering equivalent;
-    static const PartialOrdering greater;
-    static const PartialOrdering unordered;
-
-    friend constexpr bool is_eq(PartialOrdering);
-    friend constexpr bool is_lt(PartialOrdering);
-    friend constexpr bool is_gt(PartialOrdering);
-
-private:
-    enum class Type : uint8_t { Less, Equivalent, Greater, Unordered };
-    constexpr PartialOrdering(Type type) : m_type { type } { }
-    Type m_type;
-};
-constexpr bool is_eq(PartialOrdering);
-constexpr bool is_lt(PartialOrdering);
-constexpr bool is_gt(PartialOrdering);
-constexpr bool is_neq(PartialOrdering);
-constexpr bool is_lteq(PartialOrdering);
-constexpr bool is_gteq(PartialOrdering);
+// FIXME: We should remove these but std::is_eq() / std::is_neq() are not available in
+// some of our SDKs yet (rdar://87314077).
+constexpr bool is_eq(std::partial_ordering cmp) { return cmp == 0; }
+constexpr bool is_neq(std::partial_ordering cmp) { return cmp != 0; }
 
 enum TreeType { Tree, ShadowIncludingTree, ComposedTree };
 template<TreeType = Tree> ContainerNode* parent(const Node&);
 template<TreeType = Tree> Node* commonInclusiveAncestor(const Node&, const Node&);
-template<TreeType = Tree> PartialOrdering treeOrder(const Node&, const Node&);
+template<TreeType = Tree> std::partial_ordering treeOrder(const Node&, const Node&);
 
-WEBCORE_EXPORT PartialOrdering treeOrderForTesting(TreeType, const Node&, const Node&);
+WEBCORE_EXPORT std::partial_ordering treeOrderForTesting(TreeType, const Node&, const Node&);
 
 #if ASSERT_ENABLED
 
@@ -894,41 +874,6 @@ inline void Node::setTreeScopeRecursively(TreeScope& newTreeScope)
     ASSERT(!m_deletionHasBegun);
     if (m_treeScope != &newTreeScope)
         moveTreeToNewScope(*this, *m_treeScope, newTreeScope);
-}
-
-inline constexpr PartialOrdering PartialOrdering::less(Type::Less);
-inline constexpr PartialOrdering PartialOrdering::equivalent(Type::Equivalent);
-inline constexpr PartialOrdering PartialOrdering::greater(Type::Greater);
-inline constexpr PartialOrdering PartialOrdering::unordered(Type::Unordered);
-
-constexpr bool is_eq(PartialOrdering ordering)
-{
-    return ordering.m_type == PartialOrdering::Type::Equivalent;
-}
-
-constexpr bool is_lt(PartialOrdering ordering)
-{
-    return ordering.m_type == PartialOrdering::Type::Less;
-}
-
-constexpr bool is_gt(PartialOrdering ordering)
-{
-    return ordering.m_type == PartialOrdering::Type::Greater;
-}
-
-constexpr bool is_neq(PartialOrdering ordering)
-{
-    return is_lt(ordering) || is_gt(ordering);
-}
-
-constexpr bool is_lteq(PartialOrdering ordering)
-{
-    return is_lt(ordering) || is_eq(ordering);
-}
-
-constexpr bool is_gteq(PartialOrdering ordering)
-{
-    return is_gt(ordering) || is_eq(ordering);
 }
 
 inline void EventTarget::ref()

--- a/Source/WebCore/dom/Position.cpp
+++ b/Source/WebCore/dom/Position.cpp
@@ -1616,22 +1616,22 @@ std::optional<BoundaryPoint> makeBoundaryPoint(const Position& position)
     return BoundaryPoint { container.releaseNonNull(), static_cast<unsigned>(position.computeOffsetInContainerNode()) };
 }
 
-template<TreeType treeType> PartialOrdering treeOrder(const Position& a, const Position& b)
+template<TreeType treeType> std::partial_ordering treeOrder(const Position& a, const Position& b)
 {
     if (a.isNull() || b.isNull())
-        return a.isNull() && b.isNull() ? PartialOrdering::equivalent : PartialOrdering::unordered;
+        return a.isNull() && b.isNull() ? std::partial_ordering::equivalent : std::partial_ordering::unordered;
 
     auto aContainer = a.containerNode();
     auto bContainer = b.containerNode();
 
     if (!aContainer || !bContainer) {
         if (!commonInclusiveAncestor<treeType>(*a.anchorNode(), *b.anchorNode()))
-            return PartialOrdering::unordered;
+            return std::partial_ordering::unordered;
         if (!aContainer && !bContainer && a.anchorType() == b.anchorType())
-            return PartialOrdering::equivalent;
+            return std::partial_ordering::equivalent;
         if (bContainer)
-            return a.anchorType() == Position::PositionIsBeforeAnchor ? PartialOrdering::less : PartialOrdering::greater;
-        return b.anchorType() == Position::PositionIsBeforeAnchor ? PartialOrdering::greater : PartialOrdering::less;
+            return a.anchorType() == Position::PositionIsBeforeAnchor ? std::partial_ordering::less : std::partial_ordering::greater;
+        return b.anchorType() == Position::PositionIsBeforeAnchor ? std::partial_ordering::greater : std::partial_ordering::less;
     }
 
     // FIXME: Avoid computing node offset for cases where we don't need to.
@@ -1639,13 +1639,13 @@ template<TreeType treeType> PartialOrdering treeOrder(const Position& a, const P
     return treeOrder<treeType>(*makeBoundaryPoint(a), *makeBoundaryPoint(b));
 }
 
-PartialOrdering documentOrder(const Position& a, const Position& b)
+std::partial_ordering documentOrder(const Position& a, const Position& b)
 {
     return treeOrder<ComposedTree>(a, b);
 }
 
-template PartialOrdering treeOrder<ComposedTree>(const Position&, const Position&);
-template PartialOrdering treeOrder<ShadowIncludingTree>(const Position&, const Position&);
+template std::partial_ordering treeOrder<ComposedTree>(const Position&, const Position&);
+template std::partial_ordering treeOrder<ShadowIncludingTree>(const Position&, const Position&);
 
 } // namespace WebCore
 

--- a/Source/WebCore/dom/Position.h
+++ b/Source/WebCore/dom/Position.h
@@ -219,8 +219,8 @@ private:
 
 bool operator==(const Position&, const Position&);
 
-template<TreeType treeType> PartialOrdering treeOrder(const Position&, const Position&);
-WEBCORE_EXPORT PartialOrdering documentOrder(const Position&, const Position&);
+template<TreeType treeType> std::partial_ordering treeOrder(const Position&, const Position&);
+WEBCORE_EXPORT std::partial_ordering documentOrder(const Position&, const Position&);
 bool operator<(const Position&, const Position&);
 bool operator>(const Position&, const Position&);
 bool operator>=(const Position&, const Position&);

--- a/Source/WebCore/dom/Range.cpp
+++ b/Source/WebCore/dom/Range.cpp
@@ -180,7 +180,7 @@ ExceptionOr<short> Range::comparePoint(Node& container, unsigned offset) const
     auto ordering = treeOrder({ container, offset }, makeSimpleRange(*this));
     if (is_lt(ordering))
         return -1;
-    if (is_eq(ordering))
+    if (WebCore::is_eq(ordering))
         return 0;
     if (is_gt(ordering))
         return 1;
@@ -248,7 +248,7 @@ ExceptionOr<short> Range::compareBoundaryPoints(unsigned short how, const Range&
     auto ordering = treeOrder(makeBoundaryPoint(*thisPoint), makeBoundaryPoint(*otherPoint));
     if (is_lt(ordering))
         return -1;
-    if (is_eq(ordering))
+    if (WebCore::is_eq(ordering))
         return 0;
     if (is_gt(ordering))
         return 1;

--- a/Source/WebCore/dom/SimpleRange.cpp
+++ b/Source/WebCore/dom/SimpleRange.cpp
@@ -174,26 +174,26 @@ bool containsForTesting(TreeType type, const SimpleRange& range, const BoundaryP
     return false;
 }
 
-template<TreeType treeType> PartialOrdering treeOrder(const SimpleRange& range, const BoundaryPoint& point)
+template<TreeType treeType> std::partial_ordering treeOrder(const SimpleRange& range, const BoundaryPoint& point)
 {
     if (auto order = treeOrder<treeType>(range.start, point); !is_lt(order))
         return order;
     if (auto order = treeOrder<treeType>(range.end, point); !is_gt(order))
         return order;
-    return PartialOrdering::equivalent;
+    return std::partial_ordering::equivalent;
 }
 
-template<TreeType treeType> PartialOrdering treeOrder(const BoundaryPoint& point, const SimpleRange& range)
+template<TreeType treeType> std::partial_ordering treeOrder(const BoundaryPoint& point, const SimpleRange& range)
 {
     if (auto order = treeOrder<treeType>(point, range.start); !is_gt(order))
         return order;
     if (auto order = treeOrder<treeType>(point, range.end); !is_lt(order))
         return order;
-    return PartialOrdering::equivalent;
+    return std::strong_ordering::equivalent;
 }
 
-template PartialOrdering treeOrder<Tree>(const SimpleRange&, const BoundaryPoint&);
-template PartialOrdering treeOrder<Tree>(const BoundaryPoint&, const SimpleRange&);
+template std::partial_ordering treeOrder<Tree>(const SimpleRange&, const BoundaryPoint&);
+template std::partial_ordering treeOrder<Tree>(const BoundaryPoint&, const SimpleRange&);
 
 template<TreeType treeType> bool contains(const SimpleRange& outerRange, const SimpleRange& innerRange)
 {

--- a/Source/WebCore/dom/SimpleRange.h
+++ b/Source/WebCore/dom/SimpleRange.h
@@ -83,8 +83,8 @@ WEBCORE_EXPORT bool intersectsForTesting(TreeType, const SimpleRange&, const Sim
 WEBCORE_EXPORT bool intersectsForTesting(TreeType, const SimpleRange&, const Node&);
 
 // Returns equivalent if point is in range.
-template<TreeType = Tree> PartialOrdering treeOrder(const SimpleRange&, const BoundaryPoint&);
-template<TreeType = Tree> PartialOrdering treeOrder(const BoundaryPoint&, const SimpleRange&);
+template<TreeType = Tree> std::partial_ordering treeOrder(const SimpleRange&, const BoundaryPoint&);
+template<TreeType = Tree> std::partial_ordering treeOrder(const BoundaryPoint&, const SimpleRange&);
 
 struct OffsetRange {
     unsigned start { 0 };

--- a/Source/WebCore/editing/ApplyStyleCommand.cpp
+++ b/Source/WebCore/editing/ApplyStyleCommand.cpp
@@ -363,7 +363,7 @@ void ApplyStyleCommand::applyRelativeFontStyleChange(EditingStyle* style)
     // is performed in the following loops below.
     if (beyondEnd) {
         auto treeOrderPos = treeOrder(*startNode, *beyondEnd);
-        if (is_gt(treeOrderPos) || is_eq(treeOrderPos))
+        if (is_gteq(treeOrderPos))
             return;
     }
     

--- a/Source/WebCore/editing/VisiblePosition.cpp
+++ b/Source/WebCore/editing/VisiblePosition.cpp
@@ -801,7 +801,7 @@ VisiblePositionRange makeVisiblePositionRange(const std::optional<SimpleRange>& 
     return { makeContainerOffsetPosition(range->start), makeContainerOffsetPosition(range->end) };
 }
 
-PartialOrdering documentOrder(const VisiblePosition& a, const VisiblePosition& b)
+std::partial_ordering documentOrder(const VisiblePosition& a, const VisiblePosition& b)
 {
     // FIXME: Should two positions with different affinity be considered equivalent or not?
     return treeOrder<ComposedTree>(a.deepEquivalent(), b.deepEquivalent());

--- a/Source/WebCore/editing/VisiblePosition.h
+++ b/Source/WebCore/editing/VisiblePosition.h
@@ -107,7 +107,7 @@ private:
 
 bool operator==(const VisiblePosition&, const VisiblePosition&);
 
-WEBCORE_EXPORT PartialOrdering documentOrder(const VisiblePosition&, const VisiblePosition&);
+WEBCORE_EXPORT std::partial_ordering documentOrder(const VisiblePosition&, const VisiblePosition&);
 bool operator<(const VisiblePosition&, const VisiblePosition&);
 bool operator>(const VisiblePosition&, const VisiblePosition&);
 bool operator<=(const VisiblePosition&, const VisiblePosition&);

--- a/Source/WebCore/testing/Internals.cpp
+++ b/Source/WebCore/testing/Internals.cpp
@@ -6926,13 +6926,13 @@ ExceptionOr<void> Internals::setMockMediaSessionCoordinatorCommandsShouldFail(bo
 }
 #endif // ENABLE(MEDIA_SESSION)
 
-constexpr ASCIILiteral string(PartialOrdering ordering)
+constexpr ASCIILiteral string(std::partial_ordering ordering)
 {
     if (is_lt(ordering))
         return "less"_s;
     if (is_gt(ordering))
         return "greater"_s;
-    if (is_eq(ordering))
+    if (WebCore::is_eq(ordering))
         return "equivalent"_s;
     return "unordered"_s;
 }

--- a/Tools/TestWebKitAPI/Tests/WebCore/DocumentOrder.cpp
+++ b/Tools/TestWebKitAPI/Tests/WebCore/DocumentOrder.cpp
@@ -64,23 +64,23 @@ static Ref<Document> createDocument()
     return document;
 }
 
-static constexpr const char* string(PartialOrdering ordering)
+static constexpr const char* string(std::partial_ordering ordering)
 {
     if (is_lt(ordering))
         return "less";
     if (is_gt(ordering))
         return "greater";
-    if (is_eq(ordering))
+    if (WebCore::is_eq(ordering))
         return "equivalent";
     return "unordered";
 }
 
-static PartialOrdering operator-(PartialOrdering ordering)
+static std::partial_ordering operator-(std::partial_ordering ordering)
 {
     if (is_lt(ordering))
-        return PartialOrdering::greater;
+        return std::partial_ordering::greater;
     if (is_gt(ordering))
-        return PartialOrdering::less;
+        return std::partial_ordering::less;
     return ordering;
 }
 
@@ -140,7 +140,7 @@ static String join(const Vector<String>& vector, ASCIILiteral separator)
     return builder.toString();
 }
 
-static CString allPositionTypeFailures(const Position& a, Node* nodeB, unsigned offsetB, PartialOrdering expectedResult)
+static CString allPositionTypeFailures(const Position& a, Node* nodeB, unsigned offsetB, std::partial_ordering expectedResult)
 {
     Vector<String> failures;
     for (auto& b : allPositionTypes(nodeB, offsetB)) {
@@ -154,7 +154,7 @@ static CString allPositionTypeFailures(const Position& a, Node* nodeB, unsigned 
     return join(failures, ", "_s).utf8();
 }
 
-static CString allPositionTypeFailures(Node* nodeA, unsigned offsetA, Node* nodeB, unsigned offsetB, PartialOrdering expectedResult)
+static CString allPositionTypeFailures(Node* nodeA, unsigned offsetA, Node* nodeB, unsigned offsetB, std::partial_ordering expectedResult)
 {
     Vector<String> failures;
     for (auto& a : allPositionTypes(nodeA, offsetA)) {
@@ -170,25 +170,25 @@ static CString allPositionTypeFailures(Node* nodeA, unsigned offsetA, Node* node
     return join(failures, " | "_s).utf8();
 }
 
-static CString allPositionTypeFailures(Node& nodeA, unsigned offsetA, Node& nodeB, unsigned offsetB, PartialOrdering expectedResult)
+static CString allPositionTypeFailures(Node& nodeA, unsigned offsetA, Node& nodeB, unsigned offsetB, std::partial_ordering expectedResult)
 {
     return allPositionTypeFailures(&nodeA, offsetA, &nodeB, offsetB, expectedResult);
 }
 
-static CString allPositionTypeFailures(Node* nodeA, unsigned offsetA, Node& nodeB, unsigned offsetB, PartialOrdering expectedResult)
+static CString allPositionTypeFailures(Node* nodeA, unsigned offsetA, Node& nodeB, unsigned offsetB, std::partial_ordering expectedResult)
 {
     return allPositionTypeFailures(nodeA, offsetA, &nodeB, offsetB, expectedResult);
 }
 
-static CString allPositionTypeFailures(const Position& a, Node& nodeB, unsigned offsetB, PartialOrdering expectedResult)
+static CString allPositionTypeFailures(const Position& a, Node& nodeB, unsigned offsetB, std::partial_ordering expectedResult)
 {
     return allPositionTypeFailures(a, &nodeB, offsetB, expectedResult);
 }
 
 #define TEST_ALL_POSITION_TYPES(nodeA, offsetA, nodeB, offsetB, expectedResult) \
-    EXPECT_STREQ(allPositionTypeFailures(nodeA, offsetA, nodeB, offsetB, PartialOrdering::expectedResult).data(), "")
+    EXPECT_STREQ(allPositionTypeFailures(nodeA, offsetA, nodeB, offsetB, std::partial_ordering::expectedResult).data(), "")
 #define TEST_ALL_POSITION_TYPES_B(positionA, nodeB, offsetB, expectedResult) \
-    EXPECT_STREQ(allPositionTypeFailures(positionA, nodeB, offsetB, PartialOrdering::expectedResult).data(), "")
+    EXPECT_STREQ(allPositionTypeFailures(positionA, nodeB, offsetB, std::partial_ordering::expectedResult).data(), "")
 
 static Position makePositionBefore(Node& node)
 {

